### DISCUSSION TITLE
Failed to configure swap chain on creation.

### DISCUSSION THREAD

**Discussion by The-DevBlog** [bevyengine/bevy#2615] [Q&A] [answered]
I randomly started getting an error after compilation that says the following: 

```
thread 'main' panicked at 'failed to configure swap chain on creation: Swapchain can't be created for an unknown reason', C:\Users\User\.cargo\registry\src\github.com-1ecc6299db9ec823\wgpu-core-0.9.2\src\device\mod.rs:4556:29
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
error: process didn't exit successfully: `target\debug\platformer_2d.exe` (exit code: 101)
```

The game compiles fine. The game window will display for a fraction of a second and then that's when it panics and I receive this error.  I rebuilt the project and it ran fine for a while. Then the error came back. The only dependency I have in my .toml file is the Bevy main branch:

```toml
[dependencies]
 bevy = {git = "https://github.com/bevyengine/bevy", branch = "main"}
```

Does anyone have any insight on the underlying problem here?

**Comment by The-DevBlog** [comment, marked as answer]
I actually narrowed it down to having an issue with 'bevy::window::WindowMode'. I never had any issues until I switched from the cates.io bevy to the main branch.

**Comment by cpg314** [comment]
FYI, I noticed the same error happening randomly after a while on any Bevy executable, traced back to the Vukan wgpu driver. See https://github.com/gfx-rs/wgpu/issues/1948
